### PR TITLE
Add Intersection method to get UV points.

### LINKED ISSUE #1232: Add Intersection method to get UV points.
#### Description:

Add `InterstectionsAsUV` method for checking the UV parameters of intersected curves.

### PR DESCRIPTION
**Pull request by michal-pekacki**
### Issues addressed by this PR

Closes #1232 

Add `InterstectionsAsUV` method to return UV parameters of intersected curves.

### CODE REVIEW

**Comment by michal-pekacki** [comment]
@BHoMBot check compliance

**Comment by michal-pekacki** [comment]
@BHoMBot check required

File: Revit_Core_Engine/Query/Intersections.cs [hunk 1]
```diff
@@ -87,5 +87,58 @@ public static List<XYZ> Intersections(this Curve curve, CurveLoop curveLoop)
 
         /***************************************************/
 
+        [Description("UV parameters of intersection points of two curves. For no intersection, an empty list is returned.")]
+        [Input("curve1", "First curve to check the intersection for.")]
+        [Input("curve2", "Second curve to check the intersection for.")]
+        [Output("uvPoints", "List of UV parameters points.")]
+        public static List<UV> IntersectionsAsUV(this Curve curve1, Curve curve2)
```
Comment (vietle-bh) [inline Revit_Core_Engine/Query/Intersections.cs:94]: Should we give this a shorter method name of `UVIntersections`?

File: Revit_Core_Engine/Query/Intersections.cs [hunk 2]
```diff
@@ -87,5 +87,58 @@ public static List<XYZ> Intersections(this Curve curve, CurveLoop curveLoop)
...
+            return uvPoints;
+        }
+
+        /***************************************************/
+
+        [Description("UV parameters of intersection points of curve and CurveLoop. For no intersection, an empty list is returned.")]
+        [Input("curve", "Curve to check the intersection for.")]
+        [Input("curveLoop", "CurveLoop to check the intersection for.")]
+        [Output("points", "List of intersecting points.")]
+        public static List<UV> IntersectionsAsUV(this Curve curve, CurveLoop curveLoop)
```
Comment (vietle-bh) [inline Revit_Core_Engine/Query/Intersections.cs:124]: Should we give this a shorter method name of `UVIntersections`?

**Review by vietle-bh — COMMENTED** [review]
Just 2 minor comments. Well, actually 1 🤣

**Comment by michal-pekacki** [comment]
@BHoMBot check compliance

**Comment by michal-pekacki** [comment]
@BHoMBot check required

**Review by vietle-bh — APPROVED** [review]

**Comment by vietle-bh** [comment]
@BHoMBot check core
@BHoMBot check installer 
@BHoMBot check null-handling 
@BHoMBot check versioning

**Comment by vietle-bh** [comment]
@BHoMBot check versioning

**Pull request closed without merging**